### PR TITLE
feat(iot-dev): Add creation time UTC to Message properties

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Adding the creation time utc message property to D2C messages

Issue #397

### CODE REVIEW

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/Message.java [hunk 1]
```diff
@@ -605,4 +608,32 @@ public void setContentEncoding(String contentEncoding)
...
+     * ex:
+     * oct 1st, 2018 yields
+     * 2008-10-01T17:04:32.0000000
+     *
+     * @return the iot hub accepted format for the creation time utc
+     */
+    public String getCreationTimeUTCString()
+    {
+        // Codes_SRS_MESSAGE_34_064: [The function shall return the saved creationTimeUTC as a string in the format "yyyy-MM-dd_HH:mm:ss.SSSSSSS".]
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss.SSSSSSS");
```
Comment (jasmineymlo) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/Message.java]: psf?

**Comment by jasmineymlo** [comment]
e2e?

**Comment by timtay-microsoft** [comment]
> e2e?

There isn't a good e2e test to write for this scenario. Iot Hub receives this message, but there is no way for the SDK to check how the iot hub interpreted this new field.

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/Message.java (hunk 1 above)
Comment (timtay-microsoft) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/Message.java]: Sure thing

**Pull request merged**